p2p/server.py

```diff
@@ -166,10 +167,13 @@ def __init__(self,
 
     async def receive_handshake(
             self, reader: asyncio.StreamReader, writer: asyncio.StreamWriter) -> None:
+        expected_exceptions = (
+            TimeoutError, PeerConnectionLost, HandshakeFailure, asyncio.IncompleteReadError,
+            ConnectionResetError, BrokenPipeError)
         try:
             await self._receive_handshake(reader, writer)
-        except TimeoutError:
-            self.logger.debug("Timeout waiting for handshake")
+        except expected_exceptions as e:
+            self.logger.debug("Could not complete handshake: %s", e)
```

Comment:
recently I learned that you can use `self.logger.debug("msg", exc_info=True)` and it will output a fully formatted stacktrace with the log output (if you are within an `except` block)

Reply:
Nice, I'll use that